parallel-consumer-core/src/test/java/io/confluent/parallelconsumer/MockConsumerTestWithSaslAuthenticationException.java

```diff
@@ -0,0 +1,114 @@
+package io.confluent.parallelconsumer;
+
+/*-
+ * Copyright (C) 2020-2024 Confluent, Inc.
+ */
+
+import io.confluent.csid.utils.LongPollingMockConsumer;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.MockConsumer;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.clients.consumer.OffsetResetStrategy;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.errors.SaslAuthenticationException;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+import org.testcontainers.shaded.org.awaitility.Awaitility;
+
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static com.google.common.truth.Truth.assertThat;
+import static pl.tlinkowski.unij.api.UniLists.of;
+
+/**
+ * Tests that PC works fine with the plain vanilla {@link MockConsumer}, as opposed to the
+ * {@link LongPollingMockConsumer}.
+ * <p>
+ * These tests demonstrate why using {@link MockConsumer} is difficult, and why {@link LongPollingMockConsumer} should
+ * be used instead.
+ *
+ * @author Antony Stubbs
+ * @see LongPollingMockConsumer#revokeAssignment
+ */
+@Slf4j
+@Timeout(60000L)
+class MockConsumerTestWithSaslAuthenticationException {
```

Comment:
Update the comment above to be relevant to the new test.
Update / add comments in the test explaining what it is doing / expecting etc.